### PR TITLE
QICore6 Uplift for Shared Libraries

### PR DESCRIPTION
**Pull request by soy-park**
QICore v6.0.0 uplift for the following libraries: 
- QICoreCommon
- SupplementalDataElements
- Status - added USCore profiles for vital sign observations
- Hospice
- CQMCommon
- AHAOverall
- PalliativeCare
- CumulativeMedicationDuration - no changes needed in this library
- AdultOutpatientEncounters - only added `type in` for Encounters
- AdvancedIllnessandFrailty

I added the _unbundle.sh script and unbundled test cases in this PR.

### CODE REVIEW

File: input/cql/SupplementalDataElements.cql [hunk 1]
```diff
@@ -38,15 +48,22 @@ define "SDE Payer":
...
+        E.extension e
+          where e.url = 'ombCategory'
+          return e.value as Code
+      } 
+      union {
+        E.extension e
+          where e.url = 'detailed'
+          return e.value as Code
+      },
+      display: E.value
```
Comment (jbyrdevans) [inline input/cql/SupplementalDataElements.cql:64]: I think Patient.race is equivalent to Patient.extension[url = http://hl7.org/fhir/us/core/StructureDefinition/us-core-race], was there an explicit reason for this change?

I think both the "old" and the "new" implementations would result in the same value.

**Review by jbyrdevans — CHANGES_REQUESTED** [review]

File: input/cql/Status.cql [hunk 2]
```diff
@@ -16,7 +16,12 @@ context Patient
 //This library contains functions that are based on QDM 5.6 to QICore 4.1.1 March 2023 (https://github.com/cqframework/CQL-Formatting-and-Usage-Wiki/wiki/Authoring-Patterns---QICore-v4.1.1). The functions may appear similar to some QICoreCommon functions but different in that they have constraints that are relevant for measures authored by NCQA.
 
 //Assessment, Performed
-define fluent function isAssessmentPerformed(Obs List<ObservationScreeningAssessment>):
+define fluent function isAssessmentPerformed(Obs List<Choice<"SimpleObservation", 
+"LaboratoryResultObservation", 
+"ObservationClinicalResult",
+"ObservationScreeningAssessment",
+"NonPatientObservation", 
+"ObservationCancelled">>):
```
Comment (guptava) [inline input/cql/Status.cql:24]: According to the [mapping](https://hl7.org/fhir/us/qicore/STU6/qdm-to-qicore.html#assessment-performed), this should be mapped to only QI core SimpleObservation and ObservationScreening Assessment

File: input/cql/Status.cql [hunk 3]
```diff
@@ -74,7 +84,12 @@ define fluent function isProcedurePerformed(Proc List<Procedure>):
     where P.status ~ 'completed'
 
 //Laboratory Test, Performed
-define fluent function isLaboratoryTestPerformed(Obs List<LaboratoryResultObservation>):
```
Comment (guptava) [inline input/cql/Status.cql:77]: According to the [mapping](https://hl7.org/fhir/us/qicore/STU6/qdm-to-qicore.html#laboratory-test-performed), this should be Laboratory Result Observation Profile

File: input/cql/Status.cql [hunk 4]
```diff
@@ -48,7 +53,12 @@ define fluent function isLaboratoryTestOrder(ServiceRequest List<ServiceRequest>
       and S.intent = 'order'        
 
 //Diagnostic Study, Performed
-define fluent function isDiagnosticStudyPerformed(Obs List<Choice<"SimpleObservation", "LaboratoryResultObservation", "ObservationClinicalResult">>):
```
Comment (guptava) [inline input/cql/Status.cql:51]: According to the [mapping](https://hl7.org/fhir/us/qicore/STU6/qdm-to-qicore.html#diagnostic-study-performed), this should be [QICore Observation Clinical Result](https://hl7.org/fhir/us/qicore/STU6/StructureDefinition-qicore-observation-clinical-result.html) Profile

File: input/cql/Status.cql [hunk 5]
```diff
@@ -99,7 +114,12 @@ define fluent function isMedicationOrder(MedicationRequest List<MedicationReques
       and M.intent = 'order'
 
 //Physical Exam, Performed
-define fluent function isPhysicalExamPerformed(Obs List<Choice<"ObservationClinicalResult", "SimpleObservation">>):
```
Comment (guptava) [inline input/cql/Status.cql:102]: According to the [mapping](https://hl7.org/fhir/us/qicore/STU6/qdm-to-qicore.html#physical-exam-performed), this should be [QI-Core Simple Observation](https://hl7.org/fhir/us/qicore/STU6/StructureDefinition-qicore-simple-observation.html)
[QI-Core Observation Clinical Result](https://hl7.org/fhir/us/qicore/STU6/StructureDefinition-qicore-observation-clinical-result.html) Profile

File: input/cql/Status.cql [hunk 6]
```diff
@@ -124,6 +144,11 @@ define fluent function isObservationBMI(Obs List<"USCoreBMIProfile">):
     where O.status in { 'final', 'amended', 'corrected' }
 
 //Symptom
-define fluent function isSymptom(Obs List<Choice<"SimpleObservation", "ObservationScreeningAssessment">>):
```
Comment (guptava) [inline input/cql/Status.cql:127]: According to the [mapping](https://hl7.org/fhir/us/qicore/STU6/qdm-to-qicore.html#symptom), this should be [QI-Core Simple Observation](https://hl7.org/fhir/us/qicore/STU6/StructureDefinition-qicore-simple-observation.html)

**Review by guptava — COMMENTED** [review]
Added comments for Status.cql

File: input/cql/SupplementalDataElements.cql (hunk 1 above)
Comment (soy-park) [reply input/cql/SupplementalDataElements.cql]: The old implementation was showing an error

File: input/cql/Status.cql (hunk 4 above)
Comment (soy-park) [reply input/cql/Status.cql:51]: Aren't SimpleObservation, LaboratoryResultObservation, and ObservationScreeningAssessment also considered diagnostic?

File: input/cql/Status.cql (hunk 6 above)
Comment (soy-park) [reply input/cql/Status.cql:127]: Can't symptom be also considered in an ObservationClinicalResult and ObservationScreeningAssessment?

File: input/cql/SupplementalDataElements.cql (hunk 1 above)
Comment (jbyrdevans) [reply input/cql/SupplementalDataElements.cql]: Ultimately, the ICF team will be responsible for the SupplementalDataElements.cql uplift, so I think your workaround is appropriate.

**Pull request merged**